### PR TITLE
Migrate JPA FATs

### PR DESCRIPTION
**Pull request by jgrassel**
Migrating more tests to OL.  Renamed remaining buckets to follow 'com.ibm.ws.jpa.tests' project name strategy.

### CODE REVIEW

**Comment by jgrassel** [comment]
#build

File: dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAFieldInjectClient.jar/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    version="2.1">
+
+    <persistence-unit name="AppCliPU_1">
+        <provider>org.eclipse.persistence.jpa.PersistenceProvider</provider>
+        <non-jta-data-source>jdbc/JPA_NJTADS</non-jta-data-source>
+        <class>com.ibm.ws.jpa.clientcontainer.fat.basic.fieldinj.entities.SimpleEntity</class>
+        
+        <properties>
+            <!-- EclipseLink should create the database schema automatically -->
+            <property name="eclipselink.ddl-generation" value="create-tables" />
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAFieldInjectClient.jar/resources/META-INF/persistence.xml]: Are we not using the DDL generation pattern our other FATs use?

If this was a conscious decision, can we instead use:
```xml
    <property name="javax.persistence.schema-generation.database.action" value="drop-and-create"/>
    <property name="jakarta.persistence.schema-generation.database.action" value="drop-and-create"/>
```

File: dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAFieldInjectClient.jar/resources/META-INF/persistence.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+            <property name="eclipselink.target-database" value="Derby" />
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAFieldInjectClient.jar/resources/META-INF/persistence.xml]: Is it necessary to declare Derby? I'd rather these be removed if not.

File: dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAMethodInjectClient.jar/src/com/ibm/ws/jpa/clientcontainer/fat/basic/methodinj/BasicJPAMethodInjAppClient.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
+package com.ibm.ws.jpa.clientcontainer.fat.basic.methodinj;
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAMethodInjectClient.jar/src/com/ibm/ws/jpa/clientcontainer/fat/basic/methodinj/BasicJPAMethodInjAppClient.java:12]: Double check that all files have the appropriate IBM header. This is one of a couple I noticed without a header.

File: dev/com.ibm.ws.jpa_ol_fat/test-applications/defaultdatasource/resources/WEB-INF/classes/META-INF/persistence.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    <persistence-unit name="DefDS" transaction-type="JTA">
+        <provider>org.eclipse.persistence.jpa.PersistenceProvider</provider>
+
+        <class>com.ibm.ws.jpa.fat.defaultds.entity.DSEntity</class>
+        <exclude-unlisted-classes>true</exclude-unlisted-classes>
+        <properties>
+            <!-- EclipseLink should create the database schema automatically -->
+            <property name="eclipselink.ddl-generation" value="drop-and-create-tables" />
+            <property name="eclipselink.ddl-generation.output-mode"  value="both" />
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa_ol_fat/test-applications/defaultdatasource/resources/WEB-INF/classes/META-INF/persistence.xml]: Is the EclipseLink generation mode part of the test? If not, maybe we can remove this as well.

**Review by dazey3 — CHANGES_REQUESTED** [review]
Just a couple things

File: dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAFieldInjectClient.jar/resources/META-INF/persistence.xml (hunk 1 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAFieldInjectClient.jar/resources/META-INF/persistence.xml]: For these, no.  This is a quick port of a number of small tests.  I decided not to modify them to use the ddl servlet.  Sure, I can adopt drop-and-create.

File: dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAFieldInjectClient.jar/resources/META-INF/persistence.xml (hunk 2 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAFieldInjectClient.jar/resources/META-INF/persistence.xml]: Sure

File: dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAMethodInjectClient.jar/src/com/ibm/ws/jpa/clientcontainer/fat/basic/methodinj/BasicJPAMethodInjAppClient.java (hunk 3 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.container.client_fat/test-applications/BasicJPAMethodInjectClient.jar/src/com/ibm/ws/jpa/clientcontainer/fat/basic/methodinj/BasicJPAMethodInjAppClient.java:12]: THought I did, that's what reviews are for, thanks for the catch.

File: dev/com.ibm.ws.jpa_ol_fat/test-applications/defaultdatasource/resources/WEB-INF/classes/META-INF/persistence.xml (hunk 4 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa_ol_fat/test-applications/defaultdatasource/resources/WEB-INF/classes/META-INF/persistence.xml]: Yeah can get rid of that.

**Comment by jgrassel** [comment]
#build
#run-libby-bot

**Review by dazey3 — APPROVED** [review]
lg

**Comment by jgrassel** [comment]
#build
#run-libby-bot

**Comment by jgrassel** [comment]
#build
#run-libby-bot

**Pull request merged**